analyzer/QUIC.spicy

```diff
@@ -376,12 +382,21 @@ type Packet = unit(from_client: bool, context: ConnectionIDInfo&) {
   }
 
   # Peek into the header to check if it's a SHORT or LONG header
+  # and get the LONG packet type.
   : InitialByte &try {
     self.header_form = $$.initialbyte.header_form;
+    self.long_packet_type = $$.initialbyte.long_packet_type;
   }
 
   # Capture all the packet bytes if we're still have a chance of decrypting the INITIAL PACKETS
-  fpack: AllData &try if ( should_buffer(from_client, context) );
+  switch ( self.header_form == HeaderForm::LONG
+           && self.long_packet_type == LongPacketType::INITIAL
+           && should_buffer(context, from_client) ) {
+
+      # XXX: Could we do this with iterators and random access instead?
```

Comment:
Not sure I follow. As currently written you do not gain anything from a unit switch statement was you still have a single conditional field.

Reply:
Sorry, the XXX and the change away the suffixed `if ( ... )` to the switch are separate.

I think I'll just remove and go back to the suffixed `if` form. I'll remove the XXX, too.

It's mainly that the the suffixed `if` was strange to look and following doesn't work:
```
if ( some_condition ) {
   xs: bytes &size=1;
}
```
So it ended up as it is now, though that's arguably not all that better...
```
switch ( some_condition ) {
  True -> xs: bytes &size=1;
}
```